scripts/benchmark_cortex.py

```diff
@@ -1,235 +1,271 @@
+        count = await acount(prox.addNodes(self.testdata.ips))
+        assert count == self.workfactor
+        return count
+
+    @isatrial
+    async def do08LocalAddNodes(self, core: 's_cortex.Cortex', prox: 's_telepath.Client') -> int:
+        count = await acount(core.addNodes(self.testdata.asns))
+        assert count == self.workfactor
+        return count
+
+    @isatrial
+    async def do09DelNodes(self, core: 's_cortex.Cortex', prox: 's_telepath.Client') -> int:
+        count = await acount(prox.eval('inet:url | delnode'))
+        assert count == 0
+        return self.workfactor
+
+    async def run(self, core: 's_cortex.Cortex', name: str, dirn: str, coro) -> None:
+        for i in range(self.num_iters):
+            # We set up the cortex each time to avoid intra-cortex caching
+            # (there's still a substantial amount of OS caching)
+            async with self.getCortexAndProxy(dirn) as (core, prox):
+                gc.collect()
+
+                start = time.time()
+                count = await coro(core, prox)
+                self.measurements[name].append((time.time() - start, count))
+                # yappi.get_func_stats().print_all()
+
+    def _getTrialFuncs(self):
+        funcs: List[Tuple[str, Callable]] = []
+        funcnames = sorted(f for f in dir(self))
+        for funcname in funcnames:
+            func = getattr(self, funcname)
+            if not hasattr(func, '_isatrial'):
+                continue
+            funcs.append((funcname, func))
+        return funcs
+
+    async def runSuite(self, config: Dict[str, Dict], numprocs: int, tmpdir: str = None):
+        assert numprocs == 1
+        if tmpdir is not None:
+            tmpdir = os.path.abspath(tmpdir)
+        with syntest.getTestDir(tmpdir) as dirn:
+            async with self.getCortexAndProxy(dirn) as (core, _):
+                await self._loadtestdata(core)
+            for funcname, func in self._getTrialFuncs():
+                await self.run(core, funcname, dirn, func)
+
+Configs: Dict[str, Dict] = {
+    'simple': {},
+    'mapasync': {'layer:lmdb:map_async': True},
+    'dedicated': {'dedicated': True}
+}
+
+def benchmarkAll(confignames: List = None, num_procs=1, workfactor=1000, tmpdir=None) -> None:
+
+    testdata = TestData(workfactor)
+    if not confignames:
+        confignames = ['simple']
+    for configname in confignames:
+        config = Configs[configname]
+        bench = Benchmarker(config, testdata, workfactor)
+        print(f'{num_procs}-process benchmarking: {configname}')
+        asyncio.run(bench.runSuite(config, num_procs))
+        bench.report(configname)
 
 if __name__ == '__main__':
     import argparse
 
     parser = argparse.ArgumentParser()
-    parser.add_argument('which_runs', type=int, nargs='*', default=(0, 1, 2, 3))
-    parser.add_argument('--num-threads', type=int, default=1)
+    parser.add_argument('--config', nargs='*', default=None)
```

Comment:
Is this intended to be given multiple times?
```
python -m scripts.benchmark_cortex --config mapasync --config simple
```

Seems to only run the simple config

Reply:
No --config the second time. Just `--config mapasync simple`